harry-core/src/harry/dsl/HistoryBuilder.java

```diff
@@ -19,598 +19,511 @@
 package harry.dsl;
 
 import java.util.*;
-import java.util.function.Consumer;
 import java.util.function.LongSupplier;
 
-import harry.core.Run;
+import harry.core.Configuration;
+import harry.core.MetricReporter;
+import harry.ddl.SchemaSpec;
+import harry.generators.EntropySource;
+import harry.generators.JdkRandomEntropySource;
+import harry.model.Model;
 import harry.model.OpSelectors;
+import harry.model.QuiescentChecker;
+import harry.model.clock.ApproximateMonotonicClock;
+import harry.model.sut.SystemUnderTest;
+import harry.operations.Query;
+import harry.reconciler.Reconciler;
+import harry.runner.DataTracker;
 import harry.visitors.MutatingRowVisitor;
 import harry.visitors.MutatingVisitor;
 import harry.visitors.ReplayingVisitor;
 import harry.visitors.VisitExecutor;
 
-import static harry.model.OpSelectors.DefaultPdSelector.PARTITION_DESCRIPTOR_STREAM_ID;
+// TODO: we can implement a pluggable time source via a custom clock, too; the only limitation is that order
+// of timestamps has to be consistent with LTS order
 
 // TODO: we could use some sort of compact data structure or file format for navigable operation history
-public class HistoryBuilder implements Iterable<ReplayingVisitor.Visit>
+// this can have a _huge_ advantage of being able to produce arbitrary patterns, since you can specify history
+// either in a _very_ random way, or in a _very_ predictable way, but still leave a footprint extremely small,
+// since if you do not specify the details, we just produce fully random operations.
+
+/**
+ * History builder is a component for a simple yet flexible generation of arbitrary data.
+ *
+ *
```

Comment:
It seems like this is intended to be a common entrypoint for people to create new Harry tests. Those people are likely to look at this file first. Can you expand on this doc?

Reply:
Added quite a bit of doc now